### PR TITLE
Implemented MongoDB and DTO for Topic Model

### PR DESCRIPTION
**Pull request by timlenny**
Impl. the mongoDB connection for cloud storage and DTO Model for Topic

### CODE REVIEW

File: backend/src/test/java/com/timlenny/backend/service/StudyplanServiceTest.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ class StudyplanServiceTest {
 
     @Test
     void isGetAllTopicsReturningAllTopics_whenGetAllTopicsCalled() {
```
Comment (gadmel) [inline backend/src/test/java/com/timlenny/backend/service/StudyplanServiceTest.java]: What if the repo is empty? Have you tested for this case?

File: backend/src/test/java/com/timlenny/backend/controller/StudyplanControllerTest.java [hunk 2]
```diff
@@ -12,29 +8,19 @@
...
-    Topic demoTopicJava = new Topic(
-            "2", "Java", List.of(new Edge("32343232141", "1", "2")), new TopicPosition(300, 600), "Pending", "1852", 3, true
-    );
 
     @Test
     void whenGetAllTopics_theReturnListOfAllTopics() throws Exception {
-        String jsonObj = mapper.writeValueAsString(new Topic[]{demoTopicStart, demoTopicJava});
         mockMvc.perform(MockMvcRequestBuilders.get("/api/topic"))
-                .andExpect(MockMvcResultMatchers.status().isOk())
-                .andExpect(MockMvcResultMatchers.content().json(jsonObj));
```
Comment (gadmel) [inline backend/src/test/java/com/timlenny/backend/controller/StudyplanControllerTest.java:37]: Why have you removed this? Don't you want to check whether the body contents fulfills the assertion?  Maybe you want to write two separate tests for controller's success (200) for an empty repo ([]) and non-empty one ([ demoTopicStart, demoTopicJava ])?

**Review by gadmel — APPROVED** [review]
Beside the small comments, LGTM. Good job!

File: backend/src/main/java/com/timlenny/backend/model/topic/Topic.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    String id;
+    String topicName;
+    List<Edge> edges;
+    TopicPosition position;
+    String topicStatus;
+    String subtopicId;
+    int size;
+    boolean open;
+
+    public Topic(TopicDTO topicDTO) {
```
Comment (timlenny) [inline backend/src/main/java/com/timlenny/backend/model/topic/Topic.java:26]: DTO conversion according to TopicModel will be adjusted in the next step (ID, Positiom, etc. generation)

File: backend/src/test/java/com/timlenny/backend/controller/StudyplanControllerTest.java (hunk 2 above)
Comment (timlenny) [reply backend/src/test/java/com/timlenny/backend/controller/StudyplanControllerTest.java:37]: Thanks for the hint! I have adjusted the tests and covered the other test cases

File: backend/src/test/java/com/timlenny/backend/service/StudyplanServiceTest.java (hunk 1 above)
Comment (timlenny) [reply backend/src/test/java/com/timlenny/backend/service/StudyplanServiceTest.java]: Is now implemented

**Pull request merged**